src/redisearch_rs/encode_decode/tests/varint.rs

```diff
@@ -0,0 +1,262 @@
+use std::u128;
+
+use encode_decode::varint::*;
+
+#[test]
+fn test_u32() {
+    let values: [u32; 12] = [123456789, 987654321, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    let expected_lens = [4, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+
+    for (i, value) in values.iter().enumerate() {
+        let mut buf = Vec::new();
+        u32::write_as_varint(*value, &mut buf).unwrap();
+        assert_eq!(buf.len(), expected_lens[i]);
+        assert_eq!(u32::read_as_varint(&buf[..]).unwrap(), *value);
+    }
+}
+
+#[test]
+fn test_u64() {
+    let values: [u64; 12] = [123456789, 987654321, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+    let expected_lens = [4, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+
+    for (i, value) in values.iter().enumerate() {
+        let mut buf = Vec::new();
+        u64::write_as_varint(*value, &mut buf).unwrap();
+        assert_eq!(buf.len(), expected_lens[i]);
+        assert_eq!(u64::read_as_varint(&buf[..]).unwrap(), *value);
+    }
+}
+
+#[test]
+fn test_writer_error() {
+    // The buffer is too small to accommodate the encoded value.
+    let mut buf = [0u8; 1];
+    let error = u32::write_as_varint(128, buf.as_mut_slice()).unwrap_err();
+    assert_eq!(error.kind(), std::io::ErrorKind::WriteZero);
+}
+
+#[test]
+fn test_empty_reader() {
+    let error = u32::read_as_varint([].as_slice()).unwrap_err();
+    assert_eq!(error.kind(), std::io::ErrorKind::UnexpectedEof);
+}
+
+#[test]
+fn test_truncated_encoding() {
+    let mut buf = [0u8; 2];
+    let n_written_bytes = u32::write_as_varint(128, buf.as_mut_slice()).unwrap();
+    assert_eq!(n_written_bytes, 2);
+
+    let truncated = &buf[..1];
+    let error = u32::read_as_varint(truncated).unwrap_err();
+    assert_eq!(error.kind(), std::io::ErrorKind::UnexpectedEof);
+}
+
+#[test]
+/// Try to decode a number that's larger than the maximum size of the expected type.
+///
+/// The decoding process won't panic, but it'll output a non-sensical number.
+fn test_size_confusion() {
+    let mut buf = [0u8; 24];
+    let input = u128::MAX;
+    let n_written_bytes = u128::write_as_varint(input, buf.as_mut_slice()).unwrap();
+    assert_eq!(n_written_bytes, 19);
+
+    let output = u32::read_as_varint(buf.as_slice()).unwrap();
+    assert_eq!(output, 4294967295);
+    assert_ne!(output as u128, input);
+}
+
+#[test]
+fn test_u32_encoded_bytes() {
+    // Test specific values against their expected encoded byte sequences.
+    let test_cases: &[(u32, Vec<u8>)] = &[
+        (0, vec![0x00]),
+        (1, vec![0x01]),
+        // (127, vec![0x7F]),
+        // (128, vec![0x80, 0x00]),
+        // (129, vec![0x80, 0x01]),
+        // (255, vec![0x80, 0x7F]),
+        // (256, vec![0x81, 0x00]),
+        // (16383, vec![0xFE, 0x7F]),
+        // (16384, vec![0xFF, 0x00]),
```

Comment:
Not worth testing?

Reply:
Leftover while doing some experiments 😅 Fixed in [41270e9](https://github.com/RediSearch/RediSearch/pull/6346/commits/41270e9f87fa29063c51847c97c2eea8b42dc1b2)